perun/logic/temp.py

```diff
@@ -0,0 +1,653 @@
+    """
+    # Do not allow file overwrite
+    file_path = _join_to_tmp_path(file_path)
+    if os.path.exists(file_path):
+        raise exceptions.InvalidTempPathException("The temporary file '{}' already exists."
+                                                  .format(file_path))
+    # Make sure that the directory hierarchy for the file exists
+    touch_temp_dir(os.path.dirname(file_path))
+    _write_to_temp(file_path, content, json_format, protect, compress)
+
+
+def store_temp(file_path, content, json_format=False, protect=False, compress=False):
+    """Writes the 'content' to the temporary file given by the 'file_path'. The temporary file is
+    created if it does not exist and overwritten if it does.
+
+    For details regarding the path format, see the module docstring.
+
+    :param str file_path: the path to the temporary file
+    :param content: the new content of the temporary file
+    :param bool json_format: if True, the content will be formatted as json
+    :param bool protect: if True, the file will have the protected status
+    :param bool compress: if True, the content will be compressed
+    """
+    file_path = _join_to_tmp_path(file_path)
+    # Make sure that the directory hierarchy for the file exists
+    touch_temp_dir(os.path.dirname(file_path))
+    _write_to_temp(file_path, content, json_format, protect, compress)
+
+
+def get_temp(file_path):
+    """Gets the content of the temporary file 'file_path'. An exception is raised if the file does
+    not exist. None is returned if the file could not have been read, is corrupted, has
+    inconsistent or invalid index properties.
+
+    For details regarding the path format, see the module docstring.
+
+    :param str file_path: the path to the temporary file to read
+
+    :return str or None: the file content or None if an error occurred
+    """
+    # Check the existence of the file and obtain its properties
+    file_path = _join_to_tmp_path(file_path)
+    _is_tmp_file(file_path)
+    json_format, _, compressed = _get_index_entry(file_path)
+    try:
+        with open(file_path, 'rb' if compressed else 'r') as tmp_handle:
+            # Take care of possible compression
+            if compressed:
+                content = store.read_and_deflate_chunk(tmp_handle)
+            else:
+                content = tmp_handle.read()
+            # Parse the json-formatted files
+            if json_format:
+                content = json.loads(content)
+        return content
+    # Handle possible errors
+    except (OSError, ValueError, zlib.error):
+        return None
+
+
+def reset_temp(file_path):
+    """Clears the content and resets properties of the temporary file 'file_path'
+
+    For details regarding the path format, see the module docstring.
+
+    :param str file_path: the path to the temporary file
+    """
+    file_path = _join_to_tmp_path(file_path)
+    _is_tmp_file(file_path)
+    with open(file_path, 'w'):
+        pass
```

Comment:
Suggestion: This pass seems suspicious? Is this pythonic way of clearing the file?

Reply:
There are some additional options that I've found such as `truncate()` for already open files or `open(path, mode).close()` for closed files which is this case. However, the usage of `with` seemed like the best option to me.

Reply:
Ook, you can leave it like this, I would probably use the `open().close()`, but this is probably more fail-safe.